### PR TITLE
Fix some hanging issues with the repl

### PR DESCRIPTION
**Pull request by Tyilo**
This fixes the issue when the target doesn't exist or it is killed.

### CODE REVIEW

File: src/frida/repl.py [hunk 1]
```diff
@@ -78,15 +89,37 @@ def _unload_script(self):
 
         def _process_input(self, reactor):
             self._print_startup_message()
-            self._ready.wait()
+            while self._ready.wait(0.5) != True:
+                if not reactor._running:
```
Comment (oleavr) [inline src/frida/repl.py]: Should not access this variable without holding the lock; sounds like the reactor needs an accessor for that?

File: src/frida/repl.py [hunk 2]
```diff
@@ -78,15 +89,37 @@ def _unload_script(self):
...
+                    return
 
             while True:
                 expression = ""
                 line = ""
                 while len(expression) == 0 or line.endswith("\\"):
+                    if not reactor._running:
```
Comment (oleavr) [inline src/frida/repl.py]: Locking.

File: src/frida/application.py [hunk 3]
```diff
@@ -250,9 +250,10 @@ def expand_target(target):
 
 
 class Reactor(object):
-    def __init__(self, run_until_return):
+    def __init__(self, run_until_return, on_stop):
```
Comment (oleavr) [inline src/frida/application.py]: Let's not add a callback here, but instead just have the base class `ConsoleApplication` call a method that might be overridden (defaulting to doing nothing). It is after all the one calling `run()`, so it has full control.

File: src/frida/repl.py (hunk 1 above)
Comment (Tyilo) [inline src/frida/repl.py]: Why? We are only reading `_running`, which can't change from `False` to `True` at this point anyway, so when we read a `False` we know we have to stop.

Comment (oleavr) [inline src/frida/repl.py]: You are right, but this is about writing idiomatic code and avoid confusing people reading the code who will be wondering "why is this private variable accessed from outside the class, and without taking the lock?".

File: src/frida/application.py (hunk 3 above)
Comment (oleavr) [inline src/frida/application.py]: Never mind, I misremembered what the code looked like.

**Pull request merged**